### PR TITLE
Jetty 6 handler

### PR DESCRIPTION
**Pull request by achandras**

### CODE REVIEW

File: dd-java-agent/instrumentation/jetty-6/jetty-6.gradle [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+testSets {
+  latestDepTest {
+    dirName = 'test'
+  }
+}
+
+dependencies {
+  compileOnly group: 'javax.servlet', name: 'servlet-api', version: '2.5'
+  compileOnly group: 'org.mortbay.jetty', name: 'jetty', version: '6.1.26'
```
Comment (rmfitzpatrick) [inline dd-java-agent/instrumentation/jetty-6/jetty-6.gradle]: Can we confirm this w/ 6.1.0+?

File: dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/JettyHandlerAdvice.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import org.mortbay.jetty.HttpConnection;
+import org.mortbay.jetty.Response;
+
+public class JettyHandlerAdvice {
+
+  @Advice.OnMethodEnter(suppress = Throwable.class)
+  public static Scope startSpan(
+      @Advice.This final Object source, @Advice.Argument(1) final HttpServletRequest req) {
+
+    System.out.println("Starting 6 span");
```
Comment (rmfitzpatrick) [inline dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/JettyHandlerAdvice.java]: nit

File: dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/JettyHandlerAdvice.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        span.setTag(DDTags.USER_NAME, req.getUserPrincipal().getName());
+      }
+
+      Response response = HttpConnection.getCurrentConnection().getResponse();
+      DECORATE.onResponse(span, resp);
+
+      if (throwable != null) {
+        if (response.getStatus() == HttpServletResponse.SC_OK) {
+          // exception is thrown in filter chain, but status code is incorrect
+          Tags.HTTP_STATUS.set(span, 500);
```
Comment (rmfitzpatrick) [inline dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/JettyHandlerAdvice.java]: Probably best to only set the status if we're certain it's what's in the response.

File: dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/JettyHandlerAdvice.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            .extract(Format.Builtin.HTTP_HEADERS, new HttpServletRequestExtractAdapter(req));
+    final Scope scope =
+        GlobalTracer.get()
+            .buildSpan("jetty.request")
+            .asChildOf(extractedContext)
+            .withTag("span.origin.type", source.getClass().getName())
+            .startActive(false);
+
+    final Span span = scope.span();
+    final String resourceName = req.getMethod() + " " + source.getClass().getName();
```
Comment (rmfitzpatrick) [inline dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/JettyHandlerAdvice.java]: We no longer need the method in the operation name

File: dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/JettyHandlerAdvice.java (hunk 3 above)
Comment (achandras) [reply dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/JettyHandlerAdvice.java]: Removed this, but that allows weird looking situations where error tags are present with a 200 status.

File: dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/JettyHandlerAdvice.java (hunk 4 above)
Comment (achandras) [reply dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/JettyHandlerAdvice.java]: Updated in #37

File: dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/ServletHeaderAdapter.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
+package datadog.trace.instrumentation.jetty6;
```
Comment (achandras) [inline dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/ServletHeaderAdapter.java]: Added this new header extractor, as requested

File: dd-java-agent/instrumentation/jetty-6/jetty-6.gradle [hunk 6]
```diff
@@ -0,0 +1,46 @@
+// Modified by SignalFx
+muzzle {
+  pass {
+    group = "org.mortbay.jetty"
+    module = 'jetty'
+    versions = "[6.1.26,)"
```
Comment (rmfitzpatrick) [inline dd-java-agent/instrumentation/jetty-6/jetty-6.gradle]: needs correction

File: dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/JettyHandlerAdvice.java (hunk 3 above)
Comment (rmfitzpatrick) [reply dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/JettyHandlerAdvice.java]: I think removing `DECORATE.onResponse()` would prevent setting the status code unless it's known to be valid.  Is async support necessary to defer finishing until the response code is correct?

Comment (achandras) [reply dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/JettyHandlerAdvice.java]: Moved this to only be called if no exception was thrown

File: dd-java-agent/instrumentation/jetty-6/jetty-6.gradle (hunk 6 above)
Comment (achandras) [reply dd-java-agent/instrumentation/jetty-6/jetty-6.gradle]: fixed

File: dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/JettyHandlerAdvice.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+      @Advice.Thrown final Throwable throwable) {
+
+    if (scope != null) {
+      final Span span = scope.span();
+
+      if (req.getUserPrincipal() != null) {
+        span.setTag(DDTags.USER_NAME, req.getUserPrincipal().getName());
+      }
+
+      Response response = HttpConnection.getCurrentConnection().getResponse();
```
Comment (rmfitzpatrick) [inline dd-java-agent/instrumentation/jetty-6/src/main/java/datadog/trace/instrumentation/jetty6/JettyHandlerAdvice.java]: no longer used

**Pull request merged**